zapcore/sampler.go

```diff
@@ -197,12 +197,14 @@ func (s *sampler) Check(ent Entry, ce *CheckedEntry) *CheckedEntry {
 		return ce
 	}
 
-	counter := s.counts.get(ent.Level, ent.Message)
-	n := counter.IncCheckReset(ent.Time, s.tick)
-	if n > s.first && (n-s.first)%s.thereafter != 0 {
-		s.hook(ent, LogDropped)
-		return ce
+	if ent.Level >= _minLevel && ent.Level <= _maxLevel {
```

Comment:
Maybe I'm dropping here like a martian, but wouldn't it make more sense to clip the `ent.Level` to [_minLevel, _maxLevel] range? 

Will this behavior cause log entries outside valid range to bypass sampling, thus always be emitted?

Reply:
That was one of the options considered, see https://github.com/uber-go/zap/pull/975#discussion_r660809497 but there was concern that it could confuse users.

Yes, the current approach causes log entries with unknown levels to bypass sampling.

Reply:
In typical usecase for this feature would be integration with klog/logr. Those out-of-bounds entries are going to be the most common ones actually, and ones which should be sampled...probably....maybe when user actually uses those levels they are interested in all logs and will bypass sampling anyway?